x509/timestamp_cert_validations.go

```diff
@@ -33,8 +33,8 @@ func ValidateTimestampingCertChain(certChain []*x509.Certificate) error {
 	// For self-signed signing certificate (not a CA)
 	if len(certChain) == 1 {
 		cert := certChain[0]
-		if err := cert.CheckSignature(cert.SignatureAlgorithm, cert.RawTBSCertificate, cert.Signature); err != nil {
-			return fmt.Errorf("invalid self-signed certificate. subject: %q. Error: %w", cert.Subject, err)
+		if err := validateSelfSignedCert(cert, true); err != nil {
+			return err
```

Comment:
Same here, we could just add the issuer check.
```
if !bytes.Equal(cert.RawSubject, cert.RawIssuer) {
    return fmt.Errorf("self-signed certificate with subject %q is not self-issued. The issuer is %q", cert.Subject, cert.Issuer)
}
```

Reply:
Updated.